index.bs

```diff
@@ -3459,7 +3712,115 @@ The {{MLInstanceNormalizationOptions}} members are:
+
+<script type=idl>
+dictionary MLLayerNormalizationOptions {
+  MLOperand scale;
+  MLOperand bias;
+  sequence<unsigned long> axes;
+  float epsilon = 1e-5;
+};
+
+partial interface MLGraphBuilder {
+  MLOperand layerNormalization(MLOperand input, optional MLLayerNormalizationOptions options = {});
+};
+</script>
+
+The {{MLLayerNormalizationOptions}} members are:
+<dl dfn-type=dict-member dfn-for=MLLayerNormalizationOptions>
+    : <dfn>scale</dfn>
+    ::
+        An {{MLOperand}}. Specifies the N-D tensor of the scaling values whose shape is determined by the |axes| member in that each value in |axes| indicates the dimension of the input tensor with scaling values. For example, for an |axes| values of [1,2,3], the shape of this tensor is the list of the corresponding sizes of the input dimension 1, 2 and 3. When this member is not present, the scaling value is assumed to be 1.
+
+    : <dfn>bias</dfn>
+    ::
+        An {{MLOperand}}. Specifies the N-D tensor of the bias values whose shape is determined by the |axes| member in that each value in |axes| indicates the dimension of the input tensor with bias values. For example, for an |axes| values of [1,2,3], the shape of this tensor is the list of the corresponding sizes of the input dimension 1, 2 and 3. When this member is not present, the scaling value is assumed to be 0.
+
+    : <dfn>axes</dfn>
+    ::  
+        A sequence of {{unsigned long}}. The indices to the input dimensions to reduce. When this member is not present, it is assumed to be [1,2,3] that is, the reduction for the mean and variance values are calculated across all the input features for each individual sample in the batch.
+
+    : <dfn>epsilon</dfn>
+    ::
+        A {{float}} scalar. Specifies a small value to prevent computational error due to divide-by-zero.
+</dl>
+
+<div>
+    **Arguments:**
+        - *input*: an {{MLOperand}}. The input 4-D tensor.
+        - *options*: an optional {{MLLayerNormalizationOptions}}. The optional parameters of the operation.
+
+    **Returns:** an {{MLOperand}}. The layer-normalized 4-D tensor of the same shape as *input*.
+</div>
+
+<details open algorithm>
+  <summary>
+    The <dfn method for=MLGraphBuilder>layerNormalization(|input|, |options|)</dfn> method steps are:
+  </summary>
+  <div class=algorithm-steps>
+    1. [=Assert=]: the type of |input| is {{MLOperand}}.
+    1. If the [=rank=] of |input| is not 4, then [=exception/throw=] a "{{DataError}}" {{DOMException}}.
```

Comment:
> If the [=rank=] of |input| is not 4

Is this a copy-pasta typo? Like [batchNormalization](https://www.w3.org/TR/webnn/#api-mlgraphbuilder-batchnorm) and reduction, there's nothing about layerNormalization's input tensor that is dimension-specific. The visual diagrams I shared were only for the concept, and layerNormalization accepts any rank. If `axes` is unspecified, the axes just include all axes excluding dimension 0.

Forcing all callers to artificially `reshape` input to 4D and adjust the axes will complicate life for them :(, whereas handling it in the backend is trivial (assuming there was a backend that didn't accept ND directly, it would just be an update of the tensor description).

Reply:
Fixed. It is a copy-pasta typo from `InstanceNormalization`.